control-plane/connect-inject/container_init.go

```diff
@@ -107,6 +112,21 @@ func (h *Handler) containerInit(namespace corev1.Namespace, pod corev1.Pod) (cor
 		return corev1.Container{}, err
 	}
 
+	dnsEnabled, err := consulDNSEnabled(namespace, pod, h.EnableConsulDNS)
+	if err != nil {
+		return corev1.Container{}, err
+	}
+
+	var consulDNSIP string
+	if dnsEnabled {
+		for _, e := range os.Environ() {
+			if strings.Contains(e, dnsServiceHostEnvSuffix) {
+				dnsServiceHostEnv := strings.SplitN(e, "=", 2)
+				consulDNSIP = dnsServiceHostEnv[1]
+			}
+		}
+	}
```

Comment:
What if the Consul DNS service is in a different namespace? It won't be listed in the env vars.

Reply:
I don't think this is possible? connect injector and consul dns service are created in the same namespace if you're using the helm chart.

Reply:
ahh yeah sorry I thought this was the `connect-init` command